addons/account/models/reconciliation_widget.py

```diff
@@ -704,12 +704,7 @@ def _get_statement_line(self, st_line):
             'company_id': st_line.company_id.id,
         }
         if st_line.partner_id:
-            if st_line.partner_id.supplier and not st_line.partner_id.customer:
-                data['open_balance_account_id'] = st_line.partner_id.property_account_payable_id.id
-            elif not st_line.partner_id.supplier and st_line.partner_id.customer:
-                data['open_balance_account_id'] = st_line.partner_id.property_account_receivable_id.id
-            else:
-                data['open_balance_account_id'] = amount > 0 and st_line.partner_id.property_account_receivable_id.id or st_line.partner_id.property_account_payable_id.id
```

Comment:
So yeah, this part is very important (although incomplete, see https://github.com/acsone/odoo/pull/23/files).

Reply:
I agree. I've had a bugfix before because you make a move on a receivable instead of a receivable or vice versa, for instance when you do a refund. This is gonna be problematic, I guarantee it.